types/mustache/index.d.ts

```diff
@@ -1,53 +1,250 @@
-// Type definitions for Mustache 0.8.2
+// Type definitions for Mustache 0.8.3
 // Project: https://github.com/janl/mustache.js
-// Definitions by: Mark Ashley Bell <https://github.com/markashleybell>
+// Definitions by: Mark Ashley Bell <https://github.com/markashleybell>, Manuel Thalmann <https://github.com/manuth>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-
+/**
+ * A simple string scanner that is used by the template parser to find tokens in template strings.
+ */
 interface MustacheScanner {
     string: string;
     tail: string;
     pos: number;
 
+    /**
+     * Initializes a new instance of the `MustacheScanner` class.
+     */
+    new(string: string): MustacheScanner;
```

Comment:
while this compiles, it produces a really weird type. The following program is legal:

```ts
var Scanner: MustacheScanner
Scanner.pos // ???
var s = new Scanner("hi");
s.pos
var sPrime = new s("bye");  // ??!!
sPrime.pos
```

I think it should maybe be `declare class MustacheScanner` with a `constructor` instead of a `new`. From the mustache.js tests, it looks like the class doesn't have any statics.

Reply:
Well honestly I sort of copied the way to define constructors from another popular package called [`markdown-it`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/markdown-it/index.d.ts#L6).

I didn't even know this is possible :')

I'll give it a go and push my changes asap.

Thanks for the effort!